### PR TITLE
Add basic Machine Deletion docs

### PR DESCRIPTION
**Pull request by mshitrit**
Signed-off-by: Michael Shitrit <mshitrit@redhat.com>

### CODE REVIEW

File: remediation/machine-deletion/machine-deletion.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+nav_order: 3
+has_children: true
+parent: Remediation
+---
+
+# Machine-API Driven Remediation
+Generally Available
+{: .label .label-green }
+
+This operator conforms to the External Remediation of [NodeHealthCheck](https://github.com/medik8s/node-healthcheck-operator#readme) and is designed to work with [Node Health Check]((https://github.com/medik8s/node-healthcheck-operator#readme)) to reprovision unhealthy nodes using the [Machine API](https://github.com/openshift/machine-api-operator#readme). It functions by following the annotation on the Node to the associated Machine object, confirms that it has an owning controller (e.g. MachineSetController), and deletes it.  Once the Machine CR has been deleted, the owning controller creates a replacement.
```
Comment (slintes) [inline remediation/machine-deletion/machine-deletion.md]: - can you add some line breaks please for better readability in the IDE :)
- the 2nd link is broken (double braces around URL)
- NHC is mentioned twice directly after another, that doesn't sound nice IMHO ("conforms to... and is designed to")
- "It functions by following the annotation on the Node to the associated Machine object" ... what annotation is followed for doing what? 🤔 Edit: ok, after reading it a few times more I get it... but think that's an unimportant implementation detail at least at this point. Maybe just something like "it looks for the associated Machine of an unhealthy Node..."?

**Review by slintes — CHANGES_REQUESTED** [review]
I noticed that the local preview doesn't work great, links are pointing to medik8s.io instead of localhost. You can fix that by making `url` in `_config.yml` an empty string. Will send a PR for it after confirming it doesn't break the online version.
Comment inline.

File: remediation/machine-deletion/machine-deletion.md (hunk 1 above)
Comment (mshitrit) [reply remediation/machine-deletion/machine-deletion.md]: Thanks, much appreciated !

**Review by slintes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**